liferay-frontend/liferay-portal
# LPS-122818 Download svg4everybody with Gradle [SAFE TO BACKPORT]

As requested [here](https://github.com/brianchandotcom/liferay-portal/pull/95420#issuecomment-718117555).

Note that we still commit it to the repo (that way build will still work if GitHub is ever down), but we have the option of deleting the file and re-running the build in order to force a refetch.

The manual edits that I previously did are now done by a combination of a Gradle filter (to mark the `svg4everybody()` function for export) and a webpack build (this takes care of stripping out the unwanted `LEGACY_SUPPORT`).

**Test plan:**

1.  Deploy module, see it skip the download task:

    ```
    > Task :apps:frontend-js:frontend-js-svg4everybody-web:downloadSvg4Everybody SKIPPED
    ```

2.  Delete the `src/` file and see this it download again on deploy:

    ```
    > Task :apps:frontend-js:frontend-js-svg4everybody-web:downloadSvg4Everybody
    ```

    and note that `git status` is still clean (ie. no changes).

3.  In running DXP, confirm that `svg4everybody()` is still a callable global function, and other functions (eg. `getSVGAncestor()`) continue to be private. Note that this needs this to be set in portal-ext.properties:

    ```
    cdn.dynamic.resources.enabled=false
    ```
